Validation/EventGenerator/plugins/GenWeightValidation.cc

```diff
@@ -0,0 +1,177 @@
+#include "Validation/EventGenerator/interface/GenWeightValidation.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "Validation/EventGenerator/interface/DQMHelper.h"
+#include "Validation/EventGenerator/interface/HepMCValidationHelper.h"
+using namespace edm;
+
+GenWeightValidation::GenWeightValidation(const edm::ParameterSet& iPSet) : wmanager_(iPSet, consumesCollector()) {
+  genParticleToken_ = consumes<reco::GenParticleCollection>(iPSet.getParameter<edm::InputTag>("genParticles"));
+  genJetToken_ = consumes<reco::GenJetCollection>(iPSet.getParameter<edm::InputTag>("genJets"));
+  idxGenEvtInfo_ = iPSet.getParameter<int>("whichGenEventInfo");
+  idxFSRup_ = iPSet.getParameter<int>("idxFSRup");
+  idxFSRdown_ = iPSet.getParameter<int>("idxFSRdown");
+  idxISRup_ = iPSet.getParameter<int>("idxISRup");
+  idxISRdown_ = iPSet.getParameter<int>("idxISRdown");
+  leadLepPtRange_ = iPSet.getParameter<double>("leadLepPtRange");
+  leadLepPtNbin_ = iPSet.getParameter<int>("leadLepPtNbin");
+  leadLepPtCut_ = iPSet.getParameter<double>("leadLepPtCut");
+  lepEtaCut_ = iPSet.getParameter<double>("lepEtaCut");
+  rapidityRange_ = iPSet.getParameter<double>("rapidityRange");
+  rapidityNbin_ = iPSet.getParameter<int>("rapidityNbin");
+  jetPtCut_ = iPSet.getParameter<double>("jetPtCut");
+  jetEtaCut_ = iPSet.getParameter<double>("jetEtaCut");
+  nJetsNbin_ = iPSet.getParameter<int>("nJetsNbin");
+  jetPtRange_ = iPSet.getParameter<double>("jetPtRange");
+  jetPtNbin_ = iPSet.getParameter<int>("jetPtNbin");
+
+  std::vector<int> idxs = {idxFSRup_, idxFSRdown_, idxISRup_, idxISRdown_};
+  std::sort(idxs.begin(), idxs.end(), std::greater<int>());
+  idxMax_ = idxs.at(0);
+}
+
+GenWeightValidation::~GenWeightValidation() {}
+
+void GenWeightValidation::bookHistograms(DQMStore::IBooker& iBook, edm::Run const&, edm::EventSetup const&) {
+  ///Setting the DQM top directories
+  std::string folderName = "Generator/GenWeight";
+  dqm_ = new DQMHelper(&iBook);
```

Comment:
Please move to a smart pointer (e.g. `unique_ptr`)
Or, probably better: do not have dqm_ as class member, make it local and pass it by reference to the `bookTemplate` method.